exporters/common/src/main/java/io/opentelemetry/exporter/internal/TlsConfigHelper.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.internal;
+
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.X509KeyManager;
+import javax.net.ssl.X509TrustManager;
+import okhttp3.OkHttpClient;
+
+/**
+ * Utility class to help with management of TLS related components. This class is ultimately
+ * responsible for enabling TLS with the OkHttpClientBuilder by setting the sslSocketFactory. This
+ * class is only intended for internal OpenTelemetry exporter usage and should not be used by
+ * end-users.
+ */
+public class TlsConfigHelper {
+
+  private static final Logger logger = Logger.getLogger(TlsConfigHelper.class.getName());
+
+  @Nullable private X509KeyManager keyManager;
+  @Nullable private X509TrustManager trustManager;
+  @Nullable private SSLSocketFactory sslSocketFactory;
+
+  public TlsConfigHelper setTrustManager(X509TrustManager trustManager) {
+    this.trustManager = trustManager;
+    return this;
+  }
+
+  /**
+   * Creates a new X509TrustManager from the given cert content.
+   *
+   * @param trustedCertsPem Certificate in PEM format.
+   * @return this
+   */
+  public TlsConfigHelper configureTrustManager(byte[] trustedCertsPem) {
+    try {
+      this.trustManager = TlsUtil.trustManager(trustedCertsPem);
+    } catch (SSLException e) {
+      throw new IllegalStateException(
+          "Error creating trust manager for OTLP HTTP connection with provided certs. Are they valid X.509 in PEM format?",
```

Comment:
These messages should be generic of the exporter. If its important to include exporter information, let's accept it as a constructor parameter.